feat: I created recent-news-search-app with vanila JS

chl9430 opened the pull request:
## 📚 개요
뉴스 API를 이용한 최신 뉴스 검색 기능 구현

## 📚 작업사항 (기능추가/수정/삭제 등)
- [ 기능추가 ] : HTML 뼈대 만들기, 간단한 스타일 적용, 바닐라 자바스크립트로 기능 구현

## 📚 중요사항

## 📚 궁금한 점
mdn문서의 fetch와 then함수를 그냥 복사 후 붙여넣기를 해서 구현을 하였다.
이론에 대한 이해는 없고, 그냥 이렇게만 쓸 줄 알아도 되는건지 궁금하다.

## 📚 어려웠던 점

## 📚 비고

chl9430 committed:
feat: I did HTML

I created index.html

chl9430 committed:
Revert "feat: I did HTML"

This reverts commit fd2230f2cd7b82ecb673bf32151f300d3d629242.

chl9430 committed:
feat: created HTML file!

I created HTML file!

chl9430 committed:
feat: created style!

I created style file!

chl9430 committed:
feat: Done with javascript!

I have done my homework for this week!

chl9430 committed:
fix: Fixed address!

I fixed address which includes articles only related "Apple".

chl9430 committed:
feat: Clear JS, CSS files path in index.html file

I put more clear JS, CSS files path in index.html file.

danbileee commented on week6/jung-hoo/main.js:
```diff
@@ -0,0 +1,68 @@
+  resBoxEl.innerHTML = ""; // 결과를 탐색하기 전 결과물을 다 지운다.
+
+  fetch(
+    `https://newsapi.org/v2/everything?q=${inputEl.value}&sortBy=popularity&apiKey=2c157850e4c64fd99aaf77d60228d630`
```
토이프로젝트에서는 `inputEl.value` 를 그대로 써도 큰 이슈가 없겠지만,
실제 서비스라면 사용자가 입력한 값에 대한 normalize 처리가 필요할 것 같습니다.
1차로 input 엘리먼트의 maxLength나 pattern 등의 attribute를 적용하여 입력 가능한 값을 제한하고,
2차로 API 요청 전에 trim 처리 등을 추가해볼 수 있을 것 같아요!

Seok93 replied:
추가적으로 fetch, then, error 등으로 처리하다보면 콜백지옥에 빠지게 되어 가독성이 떨어지는 현상이 생길수도 있습니다. ES6부터는 이러한 비동기적 처리를 간단하게 사용하기 위해서 async/await를 제공하고 있으니 참고하여 수정해보는 것도 좋을것 같습니다.

1)[async/await](https://github.com/FC7-FE/fc7-homework/pull/11/files#diff-465bd3c7ddace5e0460470a9630245b6b7a75608ea30064c03044d7175d889f2R18-R26)를 이용하여 작성한 코드 부분 첨부
2)[async/await](https://ko.javascript.info/async-await)의 이해를 위한 사이트 첨부

danbileee commented on week6/jung-hoo/main.js:
```diff
@@ -0,0 +1,68 @@
+          // 검색 조건이 내용이라면...
+          if (
+            // 입력한 값이 기사의 내용에 포함되어 있다면...
+            data.articles[i].content
+              .toUpperCase() // 대소문자를 구분하지 않기 위해서 잠시 대문자로 바꾼다.
+              .includes(inputEl.value.toUpperCase()) // 대소문자를 구분하지 않기 위해서 잠시 대문자로 바꾼다.
```
오 저는 API 인터페이스에 제목과 내용을 받아주는 파라미터가 있는 줄 알았는데 응답값을 후처리해주셨군요!
다만 content 데이터는 꽤 긴 문자열일 것 같아서 토이프로젝트가 아닌 실무라면 API 레벨에서 처리하여 내려주는 것이 성능 우려가 없을 듯하여 만약 실무라면 서버쪽에 요청을 드려보는 게 좋겠다는 생각이 듭니다🙂

danbileee reviewed (commented):
수고하셨습니다!
충분히 고민하여 잘 구현해주신 거 같아요~
추가로 `doSearch`라는 하나의 함수 안에서 fetch함수를 chaining 하셔서 모든 처리를 해주셨는데, 모듈화하여 쪼개는 개선을 한번 해보시면 더 좋을 것 같습니다!
SOLID 원칙이라고 해서 하나의 함수는 하나의 역할만을 하도록 하는 프로그래밍 컨셉이 있는데요,
예를 들어 인풋값을 normalize하는 함수 하나, normalized된 인풋값을 받아 API 요청 후 응답값을 반환하는 함수 하나, 응답값을 받아 화면에 뉴스 결과 또는 에러메시지를 표시하는 함수 하나 이런식으로 나눌 수 있을 것 같아요!

Seok93 commented:
[코드가 깔끔해지는 3가지 원칙](https://www.youtube.com/watch?v=jafa3cqoAVM&t=4s) SOLID원칙이랑 더불어서 알고 있으면 좋은 개념들 영상 추가합니다.

chl9430 committed:
feat: Put some attributes in the input element!

I put some attributes in the input element to control what the user is going to
input.

chl9430 committed:
feat: Make it easy to read my code

I made some functions to make you guys can read my code easily.
